cmd/saml2aws/commands/login.go

```diff
@@ -79,10 +82,30 @@ func Login(loginFlags *flags.LoginExecFlags) error {
 
 	log.Printf("Authenticating as %s ...", loginDetails.Username)
 
-	samlAssertion, err := provider.Authenticate(loginDetails)
-	if err != nil {
-		return errors.Wrap(err, "error authenticating to IdP")
+	var samlAssertion string
+	if account.SAMLCache {
+		if valid, err := cacheProvider.IsValid(); valid {
+			samlAssertion, err = cacheProvider.Read()
+			if err != nil {
+				logger.Debug("Could not read cache:", err)
```

Comment:
There probably should be a `ErrNotFound` rather than just skipping all errors and continuing, kinda makes the error redundant if you always ignore it.

Reply:
good call, i'll actually error on read failure